python/pyspark/pandas/series.py

```diff
@@ -5179,7 +5179,9 @@ def asof(self, where: Union[Any, List]) -> Union[Scalar, "Series"]:
 
         Notes
         -----
-        Indices are assumed to be sorted. Raises if this is not the case.
```

Comment:
I am curious about why `ps.Series.asof`'s behavior is not same with Pandas https://pandas.pydata.org/docs/reference/api/pandas.Series.asof.html

Reply:
when values of the series is not sorted
```python
>>> pser = pd.Series([2, 1, np.nan, 4], index=[10, 20, 30, 40])
>>> psser = ps.from_pandas(pser)
>>> pser.asof(20)
1.0
>>> psser.asof(20)
2.0
>>>
```

Reply:
Looks like a bug of `asof`, it should be equal I think.

Reply:
Yeah we should fix

Reply:
[SPARK-37491](https://issues.apache.org/jira/browse/SPARK-37491) was added to track this bug, @itholic FYI

Reply:
Thanks :-)